### PR TITLE
Refactor summary collector

### LINKED ISSUE #2735: Improve performance of summary_collector
## Issue

In SummaryCollector.loadAllSummaryData a numpy array is built by copying  numerical values and active mask individually via the Python /C++ interop is slow when there is a large number of elements

## Suggested change

Building the returning  numpy array completely in C++ and returning using PyBind will result improved performance 

For reference see implementation for #2674

### PR DESCRIPTION
**Pull request by frode-aarstad**
**Issue**
Resolves #2735

**Approach**
Improves the performance by moving more of the code filling out the summary dataframe from python back to c++.

## Pre review checklist

- [x] Added appropriate labels

Adding labels helps the maintainers when writing release notes, see sections and the
corresponding labels here: https://github.com/equinor/ert/blob/main/.github/release.yml

### CODE REVIEW

File: libres/lib/include/ert/python.hpp [hunk 1]
```diff
@@ -53,6 +53,7 @@ struct Submodule {
 namespace py_conversion {
 
 enkf_main_type *to_enkf_main_type(py::object obj);
+enkf_main_type *to_enkfmain_type(py::object obj);
```
Comment (frode-aarstad) [inline libres/lib/include/ert/python.hpp:56]: Something strange going on here. The `to_enkf_main_type` function returns a EnkfFsManager but casts to a enkf_main_type?? 

I had to add a function that actually returns a enkf_main_type but need a proper name for it. Perhaps its better to do some renaming

Comment (TerryHannant) [reply libres/lib/include/ert/python.hpp:56]: Maybe @oyvindeide can shed some light on this?

Comment (oyvindeide) [reply libres/lib/include/ert/python.hpp:56]: Agree that this looks a bit strange, but is a result how cwrap does the binding. Both are `enkf_main_type` as there is no `enkf_fs_manager_type`. Maybe @pinkwah has a better solution to converting the BaseCClass to the corresponding type though?

Comment (frode-aarstad) [reply libres/lib/include/ert/python.hpp:56]: I will leave this for now and create an issue for later.

File: res/enkf/export/summary_collector.py [hunk 2]
```diff
@@ -50,32 +49,16 @@ def loadAllSummaryData(ert: EnKFMain, case_name, keys=None, realization_index=No
...
+        data_array = _lib.enkf_fs_summary_data.get_summary_data(
+            ensemble_config, fs, summary_keys, realizations, len(dates)
+        )
 
         multi_index = MultiIndex.from_product(
             [realizations, dates], names=["Realization", "Date"]
         )
         summary_data = DataFrame(
-            data=numpy.transpose(summary_array), index=multi_index, columns=summary_keys
+            data=numpy.transpose(data_array), index=multi_index, columns=summary_keys
```
Comment (TerryHannant) [inline res/enkf/export/summary_collector.py:62]: If you change the ordering of population of the data array in C++ then the transpose is no longer needed

**Review by TerryHannant — APPROVED** [review]
Look good

**Pull request merged**